common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -352,6 +360,51 @@ protected void validateTokenRequest(MicrosoftStsTokenRequest request) {
         //TODO implement
     }
 
+    @Override
+    protected HttpResponse performTokenRequest(final MicrosoftStsTokenRequest request) throws IOException {
+        final String methodName = ":performTokenRequest";
+
+        Logger.verbose(
+                TAG + methodName,
+                "Performing token request..."
+        );
+
+        final String requestBody = ObjectMapper.serializeObjectToFormUrlEncoded(request);
+        final Map<String, String> headers = new TreeMap<>();
+        headers.put("client-request-id", DiagnosticContext.getRequestContext().get(DiagnosticContext.CORRELATION_ID));
+        headers.putAll(Device.getPlatformIdParameters());
+
+        HttpResponse response =  HttpRequest.sendPost(
+                new URL(mTokenEndpoint),
+                headers,
+                requestBody.getBytes(ObjectMapper.ENCODING_SCHEME),
+                TOKEN_REQUEST_CONTENT_TYPE
+        );
```

Comment:
From here up appears to be copy/paste from the `super` class -- why not call `HttpResponse response = super.performTokenRequest()` ?

Reply:
addressed